src/ZosJobsProvider.ts

```diff
@@ -280,75 +282,114 @@ export class ZosJobsProvider implements vscode.TreeDataProvider<Job> {
+                    }
                 }
-                searchCriteria = response.label;
             }
-            if (searchCriteria === ZosJobsProvider.defaultDialogText) {
+            if (!searchCriteria) { // Do we have anything to search with yet?
+                // if (searchCriteria === ZosJobsProvider.defaultDialogText) {
+                let options: vscode.InputBoxOptions;
                 let owner: string;
                 let prefix: string;
                 let jobid: string;
                 // manually entering a search
-                let options: vscode.InputBoxOptions = {
-                    prompt: localize("jobsFilterPrompt.option.prompt.owner",
-                    "Search Jobs by entering an Owner and/or a Prefix or a JobID on it's own.\n\nEnter an Owner default is * to include all."),
-                    value: node.owner
-                };
-                // get user input
-                owner = await vscode.window.showInputBox(options);
-                if (!owner) {
-                    vscode.window.showInformationMessage(localize("jobsFilterPrompt.enterOwner",
-                            "No valid value for owner or wild card *. Search Cancelled"));
-                    return;
-                }
-                owner = owner.toUpperCase();
-                options = {
-                    prompt: localize("jobsFilterPrompt.option.prompt.prefix", "Now enter a Job prefix default is * to include all jobs."),
-                    value: node.prefix
-                };
-                // get user input
-                prefix = await vscode.window.showInputBox(options);
-                if (!searchCriteria) {
-                    vscode.window.showInformationMessage(localize("jobsFilterPrompt.enterPrefix",
-                            "No valid value for prefix or wild card *. Search Cancelled"));
-                    return;
-                }
-                prefix = prefix.toUpperCase();
-                options = {
-                    prompt: localize("jobsFilterPrompt.option.prompt.jobid", "\nEnter a specific Job id or leave blank"),
-                    value: node.searchId
-                };
-                // get user input
-                jobid = await vscode.window.showInputBox(options);
-                if (!searchCriteria) {
-                    vscode.window.showInformationMessage(localize("jobsFilterPrompt.enterPrefix", "Search Cancelled"));
-                    return;
+                if (!hasHistory || choice === this.createOwner) { // User has selected owner/prefix option
+                    options = {
+                        prompt: localize("jobsFilterPrompt.option.prompt.owner",
+                        "Enter the Job Owner. Default is *."),
+                        value: node.owner
+                    };
+                    // get user input
+                    owner = await vscode.window.showInputBox(options);
+                    if (owner === undefined) {
```

Comment:
I assume this is how VSCode APIs report a cancelled/escaped action. otherwise it will be empty or null?

Reply:
Exactly. If I checked for !owner I would be including if the user specific an empty string on purpose